tools/confix/migrations.go

```diff
@@ -19,19 +19,44 @@ const (
 )
 
 // MigrationMap defines a mapping from a version to a transformation plan.
-type MigrationMap map[string]func(from *tomledit.Document, to, planType string) transform.Plan
+type MigrationMap map[string]func(from *tomledit.Document, to, planType string) (transform.Plan, *tomledit.Document)
 
 var Migrations = MigrationMap{
-	"v0.45": NoPlan, // Confix supports only the current supported SDK version. So we do not support v0.44 -> v0.45.
-	"v0.46": PlanBuilder,
-	"v0.47": PlanBuilder,
-	"v0.50": PlanBuilder,
-	"v0.52": PlanBuilder,
+	"v0.45":    NoPlan, // Confix supports only the current supported SDK version. So we do not support v0.44 -> v0.45.
+	"v0.46":    PlanBuilder,
+	"v0.47":    PlanBuilder,
+	"v0.50":    PlanBuilder,
+	"v0.52":    PlanBuilder,
+	"serverv2": V2PlanBuilder,
 	// "v0.xx.x": PlanBuilder, // add specific migration in case of configuration changes in minor versions
 }
 
+type v2KeyChangesMap map[string][]string
+
+var v2KeyChanges = map[string]v2KeyChangesMap{
```

Comment:
nit, given that it is v2 specific. why do we need a map of map?
I feel like `map[string][]string` is sufficient

Reply:
I thought we might need it when we have updated versions related to serverv2 config in the future.

Reply:
I think for serverv2 -> serverv2.1 we'll be able to use the default PlanBuilder, so this may not be needed